### DISCUSSION TITLE
Make it possible to set WEBHOME and WEBPUBLIC differently.

### DISCUSSION THREAD

**Discussion by illmaren** [shinsenter/php#36] [Ideas]
For Frameworks like Laravel this will not work without changing the Image. 
If you set WEBHOME to /var/www/public you cant access any files above the public folder.
But if you set it to /var/www the Laravel installation works but you need to go over http://example.com/public to get to the laravel page.

Just a small addition to make it possible

**Comment by shinsenter** [comment]
@illmaren 

You can use the `WEBHOME` environment variable to specify the web root directory of the web server, and the `APACHE_DOCUMENT_ROOT` or `NGINX_DOCUMENT_ROOT` environment variable to specify the directory containing the index.php file inside your web root directory.

### PHP-FPM + Apache
- [![`shinsenter/phpfpm-apache`](https://img.shields.io/docker/image-size/shinsenter/phpfpm-apache/latest?label=shinsenter%2Fphpfpm-apache)](https://docker.shin.company/phpfpm-apache)

```
ARG  PHP_VERSION=8.2
FROM shinsenter/phpfpm-apache:php${PHP_VERSION}

# ==========================================================

# Sets the directory from which Apache will serve files
ENV WEBHOME="/var/www/html"
ENV APACHE_DOCUMENT_ROOT="/public"
```

### PHP-FPM + Nginx
- [![`shinsenter/phpfpm-nginx`](https://img.shields.io/docker/image-size/shinsenter/phpfpm-nginx/latest?label=shinsenter%2Fphpfpm-nginx)](https://docker.shin.company/phpfpm-nginx)

```
ARG  PHP_VERSION=8.2
FROM shinsenter/phpfpm-nginx:php${PHP_VERSION}

# ==========================================================

# Sets the directory from which Nginx will serve files
ENV WEBHOME="/var/www/html"
ENV NGINX_DOCUMENT_ROOT="/public"
```
---

I also added various docker images for popular PHP open source projects:

- [![`shinsenter/cakephp4`](https://img.shields.io/docker/image-size/shinsenter/cakephp4/latest?label=shinsenter%2Fcakephp4)](https://docker.shin.company/cakephp4)
- [![`shinsenter/codeigniter4`](https://img.shields.io/docker/image-size/shinsenter/codeigniter4/latest?label=shinsenter%2Fcodeigniter4)](https://docker.shin.company/codeigniter4)
- [![`shinsenter/crater`](https://img.shields.io/docker/image-size/shinsenter/crater/latest?label=shinsenter%2Fcrater)](https://docker.shin.company/crater)
- [![`shinsenter/flarum`](https://img.shields.io/docker/image-size/shinsenter/flarum/latest?label=shinsenter%2Fflarum)](https://docker.shin.company/flarum)
- [![`shinsenter/fuelphp`](https://img.shields.io/docker/image-size/shinsenter/fuelphp/latest?label=shinsenter%2Ffuelphp)](https://docker.shin.company/fuelphp)
- [![`shinsenter/grav`](https://img.shields.io/docker/image-size/shinsenter/grav/latest?label=shinsenter%2Fgrav)](https://docker.shin.company/grav)
- [![`shinsenter/hyperf`](https://img.shields.io/docker/image-size/shinsenter/hyperf/latest?label=shinsenter%2Fhyperf)](https://docker.shin.company/hyperf)
- [![`shinsenter/kirby`](https://img.shields.io/docker/image-size/shinsenter/kirby/latest?label=shinsenter%2Fkirby)](https://docker.shin.company/kirby)
- [![`shinsenter/laminas`](https://img.shields.io/docker/image-size/shinsenter/laminas/latest?label=shinsenter%2Flaminas)](https://docker.shin.company/laminas)
- [![`shinsenter/laravel`](https://img.shields.io/docker/image-size/shinsenter/laravel/latest?label=shinsenter%2Flaravel)](https://docker.shin.company/laravel)
- [![`shinsenter/mautic`](https://img.shields.io/docker/image-size/shinsenter/mautic/latest?label=shinsenter%2Fmautic)](https://docker.shin.company/mautic)
- [![`shinsenter/phpmyadmin`](https://img.shields.io/docker/image-size/shinsenter/phpmyadmin/latest?label=shinsenter%2Fphpmyadmin)](https://docker.shin.company/phpmyadmin)
- [![`shinsenter/symfony`](https://img.shields.io/docker/image-size/shinsenter/symfony/latest?label=shinsenter%2Fsymfony)](https://docker.shin.company/symfony)
- [![`shinsenter/slim`](https://img.shields.io/docker/image-size/shinsenter/slim/latest?label=shinsenter%2Fslim)](https://docker.shin.company/slim)
- [![`shinsenter/statamic`](https://img.shields.io/docker/image-size/shinsenter/statamic/latest?label=shinsenter%2Fstatamic)](https://docker.shin.company/statamic)
- [![`shinsenter/wordpress`](https://img.shields.io/docker/image-size/shinsenter/wordpress/latest?label=shinsenter%2Fwordpress)](https://docker.shin.company/wordpress)
- [![`shinsenter/yii`](https://img.shields.io/docker/image-size/shinsenter/yii/latest?label=shinsenter%2Fyii)](https://docker.shin.company/yii)

**Reply by shinsenter** [reply]
@illmaren 

I am making more clearer documentation about settings of application directory and document root directory for my new docker images.

<img width="805" alt="Screenshot 2024-02-02 at 8 39 33" src="https://github.com/shinsenter/php/assets/2082119/28eab0d6-ac39-4992-9a19-29a7d55b5e49">

https://hub.docker.com/r/shinsenter/php
